static/app/views/eventsV2/table/quickContext.tsx

```diff
@@ -78,6 +96,22 @@ function getHoverHeader(dataRow: EventData, contextType: ContextType) {
   ) : null;
 }
 
+const addFieldAsColumn = (fieldName: string, location?: Location) => {
```

Comment:
One thing I noticed is that adding a field as a column breaks if we're on a saved query, that's because when there aren't any changes we make a short URL that doesn't actually include the fields in the query params. So adding a field here: https://localhost:7999/organizations/sentry/discover/results/?id=17426&statsPeriod=24h

causes all the fields to get wiped and replaced with the new field. I think we'll need to pass the saved query as well and do something like `location?.query.field ?? savedQuery.fields`

Reply:
err, you'll likely be able to grab these from `eventView` actually, savedQuery isn't passed down to the table.

Also, since this is a functional component you _could_ avoid passing location down and use the `useLocation` hook

Reply:
Fixed!

Reply:
I believe theres also useOrganization and useProjects if you want to use those too